rest/route/host.go

```diff
@@ -80,16 +85,9 @@ func (h *hostsChangeStatusesHandler) Run(ctx context.Context) gimlet.Responder {
 		if err != nil {
 			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding host '%s' with owner '%s'", id, user.Id))
 		}
-		switch status.Status {
-		case evergreen.HostTerminated:
-			err = errors.WithStack(cloud.TerminateSpawnHost(ctx, evergreen.GetEnvironment(), foundHost, user.Id, "terminated via REST API"))
-		case evergreen.HostQuarantined:
-			err = units.DisableAndNotifyPoisonedHost(ctx, evergreen.GetEnvironment(), foundHost, false, fmt.Sprintf("quarantined via REST API by user '%s'", user.Id), user.Id)
-		default:
-			err = foundHost.SetStatus(ctx, status.Status, user.Id, fmt.Sprintf("changed by %s from API", user.Id))
-		}
+		_, _, err = api.ModifyHostStatus(ctx, h.env, foundHost, status.Status, "", user)
```

Comment:
For the note, can we pass in a non-empty string like `"modified by REST API"`? That would help distinguish that the host status was changed by the REST route rather than the UI.